Source/ARTRealtime.m

```diff
@@ -776,20 +794,16 @@ - (void)onConnected:(ARTProtocolMessage *)message {
     
     // Resuming
     if (_resuming) {
-        if (![message.connectionId isEqualToString:self.connection.id_nosync]) { // RTN15c3
-            [self.logger warn:@"RT:%p connection \"%@\" has reconnected, but resume failed. Reattaching any attached channels", self, message.connectionId];
-            // Reattach all channels
-            for (ARTRealtimeChannelInternal *channel in self.channels.nosyncIterable) {
-                [channel reattachWithReason:message.error callback:nil];
-            }
-            _resuming = false;
-        }
-        else if (message.error) {
-            [self.logger warn:@"RT:%p connection \"%@\" has resumed with non-fatal error \"%@\"", self, message.connectionId, message.error.message];
-            // The error will be emitted on `transition`
-        }
-        else {
+        if ([message.connectionId isEqualToString:self.connection.id_nosync] && !message.error) {  // RTN15c6
+            [self reattachChannelsOnResumeResult:message];
```

Comment:
I don't understand what the `_resuming` flag is doing here - we need to reattach channels any time we get a new connection right? In which case is it better to move `reattachChannelsOnResumeResult` out of these `if/else` blocks to make it explicit we will always reattach on a new connection?

Reply:
In this context it is a local flag set after connection went into `suspended`, `disconnected` or `failed` state telling the client that connected signal arrived after those event happened. I got it as" apply this code block from 'resume' section of the feature spec". To be honest I do not like this method of tracking and modifying state like this too. But I didn't want to remove this flag in risk to collapsing code that would potentially break other specs that was outside of scope of this task. 

Do you think this is not the correct behaviour? I quickly turned off the flag and can observe that lots of other tests are failing now - I'd assume that this is necessary so that we differentiate between the first connections and subsequent ones

Reply:
For me the main thing is whenever we get a `CONNECTED` response we **always** reattach all channels, regardless of what is in the `CONNECTED` message (eg it doesn't matter if the connection ID changed or if we got an error). Could the reattach be moved out of the `if/else` to make sure?

eg in ably-js we reattach whenever we get a new transport activated (which only occurs after a `CONNECTED`) https://github.com/ably/ably-js/pull/1045/commits/66e95d01b62672e0815955f9b1f0ee4c493328e6

Reply:
9a278d2f875d79113121aac6300c575ca41fc0bd gets rid of _resuming that should address this as well